ddtrace/contrib/mariadb/patch.py

```diff
@@ -0,0 +1,50 @@
+import mariadb
+from ddtrace import Pin
+from ddtrace import config
+from ddtrace.contrib.dbapi import TracedConnection
+from ddtrace.vendor import wrapt
+
+from ...ext import db
+from ...ext import net
+
+from ...utils.formats import asbool
+from ...utils.formats import get_env
+
+
+config._add(
+    "mariadb",
+    dict(
+        trace_fetch_methods=asbool(get_env("mariadb", "trace_fetch_methods", default=False)),
+        _default_service="mariadb",
+    ),
+)
+
+
+
+
+def patch():
+    wrapt.wrap_function_wrapper("mariadb", "connect", _connect)
```

Comment:
we want to add a guard here to ensure we don't double patch mariadb.

https://github.com/DataDog/dd-trace-py/blob/0dee2fcc7ecbdde7bb95b015b66b638660e349ef/ddtrace/contrib/redis/patch.py#L26-L28

We set the attribute on the module instead of just checking if it is already an ObjectProxy because if some other vendor has wrapped it it might already be an ObjectProxy, in those cases we still want to wrap it.